### PR TITLE
Avoid cloud jump when switching between mainmenu and loading screen

### PR DESCRIPTION
**Pull request by grorp**
This PR prevents the clouds from jumping when switching between mainmenu and loading screen by using the same Clouds object for both.

The mainmenu clouds already used shaders before. I had to choose between both or neither, so now both the mainmenu clouds and the loading screen clouds use shaders if available.

## To do

This PR is a Ready for Review.

## How to test

Look at clouds in the mainmenu and on the loading screen, verify that they work with and without shaders.

See that there's no cloud jump anymore when switching between mainmenu and loading screen.

### CODE REVIEW

File: src/client/clouds.cpp [hunk 1]
```diff
@@ -28,11 +28,10 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "settings.h"
 #include <cmath>
 
-
-// Menu clouds are created later
 class Clouds;
-Clouds *g_menuclouds = NULL;
+IWritableShaderSource *g_menu_shader_source = NULL;
```
Comment (sfan5) [inline src/client/clouds.cpp:32]: this can be moved to clientlauncher.cpp and doesn't need to be public
also `unique_ptr`

Comment (grorp) [reply src/client/clouds.cpp]: I tried unique_ptr, but I still had to delete the thing manually to get correct deletion order and it didn't work with forward-declaring IWritableShaderSource in clientlauncher.h

Comment (sfan5) [reply src/client/clouds.cpp]: but you don't need to declare it in clientlauncher.h?

Comment (grorp) [reply src/client/clouds.cpp]: Ah, I see

**Review by sfan5 — APPROVED** [review]
works

**Comment by siliconsniffer** [comment]
Very nice addition. Such a small detail, but it really makes a difference. Well done!

**Pull request merged**